### PR TITLE
Update user guide to add how to handle `permission issue`

### LINKED ISSUE #325: Update `user guide` documentation to include instructions on how to deal with npm install permission issues
We should update [`Getting started`](https://github.com/sonarwhal/sonar/blob/47ceeea2555a9867554b4cd96d5cb746201035b5/docs/user-guide/index.md#getting-started) to contain contain some information on how to deal with [`npm install` permission issues](https://github.com/sonarwhal/sonar/issues/308#issuecomment-311237433).

---

See also: https://github.com/sonarwhal/sonar/issues/308

### PR DESCRIPTION
**Pull request by qzhou1607**
Fix #325

### CODE REVIEW

File: docs/user-guide/index.md [hunk 1]
```diff
@@ -57,6 +57,9 @@ the different pieces:
 * [Connectors](#connectors)
 * [Formatters](#formatters)
 
+## Permission Issue
```
Comment (alrra) [inline docs/user-guide/index.md]: Add empty line after this line

(see also: https://travis-ci.org/sonarwhal/sonar/jobs/250504088#L2417)

File: docs/user-guide/index.md [hunk 2]
```diff
@@ -57,6 +57,9 @@ the different pieces:
...
+If you receive an `EACCES` error when installing `sonar`, the recommended solution is to [change npm's default directory](https://docs.npmjs.com/getting-started/fixing-npm-permissions#option-2-change-npms-default-directory-to-another-directory) and then try again. So far, one such permission issue has been reported when user tries to install `sonar` on [Windows Subsystem for Linux](https://msdn.microsoft.com/en-us/commandline/wsl/install_guide). Dependency `canvas-prebuilt` throws an `EACCES` error during the installation process, and this [issue](https://github.com/sonarwhal/sonar/issues/308) was resolved adopting the recommended solution. You can find detailed steps on how to change the npm default directory [here](https://docs.npmjs.com/getting-started/fixing-npm-permissions#option-2-change-npms-default-directory-to-another-directory).
```
Comment (alrra) [inline docs/user-guide/index.md]: Split this line into multiple shorter lines, and were possible try to keep the length of the lines to under 80 characters.

File: docs/user-guide/index.md (hunk 1 above)
Comment (qzhou1607) [inline docs/user-guide/index.md]: @alrra Done. One question - some links can get quite long in markdown. Do we always have to use inline links? Referenced links seem to work better in those situations.

File: docs/user-guide/index.md [hunk 3]
```diff
@@ -57,6 +57,16 @@ the different pieces:
...
 
+## Permission Issue
+
+If you receive an `EACCES` error when installing `sonar`, the recommended
+solution is to [change npm's default directory](https://docs.npmjs.com/getting-started/fixing-npm-permissions#option-2-change-npms-default-directory-to-another-directory) and then try again. So far, one such permission issue has been reported
+when user tries to install `sonar` on [Windows Subsystem for Linux](https://msdn.microsoft.com/en-us/commandline/wsl/install_guide). Dependency
+`canvas-prebuilt` throws an `EACCES` error during the installation process,
+and this [issue](https://github.com/sonarwhal/sonar/issues/308) was resolved
+adopting the recommended solution. You can find detailed steps on how to change
+the npm default directory [here](https://docs.npmjs.com/getting-started/fixing-npm-permissions#option-2-change-npms-default-directory-to-another-directory).
```
Comment (molant) [inline docs/user-guide/index.md]: I was also capable of reproducing this with Mac.
I had node installed via the installer in their website but it looks like if you use [Homebrew](https://brew.sh/) that problem doesn't show up (didn't had the time to test).
@alrra how did you installed node in your machine?

File: docs/user-guide/index.md (hunk 1 above)
Comment (alrra) [inline docs/user-guide/index.md]: > some links can get quite long in markdown

Yes, that's why I said _"**were possible** try to keep the length..."_

> Referenced links seem to work better in those situations.

Can you give an example on how that would look like? Thanks!

File: docs/user-guide/index.md (hunk 3 above)
Comment (alrra) [inline docs/user-guide/index.md]: > @alrra how did you installed node in your machine?

I use [nvm](https://github.com/creationix/nvm) both on macOS and Bash on Windows.

Comment (molant) [inline docs/user-guide/index.md]: So maybe it is only something that happens if you use the official installer?

File: docs/user-guide/index.md (hunk 1 above)
Comment (qzhou1607) [inline docs/user-guide/index.md]: @alrra in the actual text part
```
...the recommended
solution is to [change npm's default directory][1] and then try again...
```
and then append the following at the end of the markdown
```
[1]:https://docs.npmjs.com/getting-started/fixing-npm-permissions#option-2-change-npms-default-directory-to-another-directory
```

Comment (alrra) [inline docs/user-guide/index.md]: @qzhou1607 If there are no problems when converting it to HTML, I'm ok with that.

@sonarwhal/contributors Other opinions?

Comment (qzhou1607) [inline docs/user-guide/index.md]: @alrra checked in Hexo, looks okay. I've pushed up the changes.

File: docs/user-guide/index.md (hunk 3 above)
Comment (qzhou1607) [inline docs/user-guide/index.md]: @molant Added more information.

**Pull request closed without merging**